### PR TITLE
build+chain: replace pebbe/zmq4 with lightninglabs/gozmq (native Go)

### PR DESCRIPTION
**Pull request by aakselrod**
This PR replaces a ZMQ binding that depends on C libraries with a pure Go library to keep cross-compilation easy.

### CODE REVIEW

File: chain/bitcoind.go [hunk 1]
```diff
@@ -511,40 +512,8 @@ func (c *BitcoindClient) socketHandler() {
...
-	}
-	err = zmqClient.SetSubscribe("rawtx")
-	if err != nil {
-		log.Error(err)
-		return
-	}
-	err = zmqClient.SetRcvtimeo(c.zmqPollInterval)
-	err = zmqClient.Connect(c.zmqConnect)
+	zmqClient, err := gozmq.Subscribe(c.zmqConnect, []string{"rawblock",
+		"rawtx"}, c.zmqPollInterval)
```
Comment (Roasbeef) [inline chain/bitcoind.go:516]: We don't to worry about closing the socket or anything like that?

Comment (aakselrod) [reply chain/bitcoind.go:516]: Good point, forgot to add that. Thanks!

Comment (aakselrod) [reply chain/bitcoind.go:516]: Fixed.

**Review by Roasbeef — APPROVED** [review]
LGTM 💡

**Pull request merged**